skops/hub_utils/_hf_hub.py

```diff
@@ -561,7 +571,10 @@ def download(
     dst = Path(dst)
     if dst.exists() and bool(next(dst.iterdir(), None)):
         raise OSError("None-empty dst path already exists!")
-    dst.rmdir()
+
+    # remove the folder only if it's empty and it exists
+    if dst.exists():
```

Comment:
I'm a bit astonished that the missing check did not cause trouble so far.

Reply:
same